test/ui/SimpleFilterTest.ts

```diff
@@ -0,0 +1,181 @@
+      test = null;
+    });
+
+    it('should set the title correctly', () => {
+      expect(test.cmp.options.title).toBe('FooTitleBar');
+    });
+
+    it('should set the field correctly', () => {
+      expect(test.cmp.options.field).toBe('@field');
+    });
+
+    it('should allow to getValueCaption', () => {
+      test.cmp.options.field = '@filetype';
+      expect(test.cmp.getValueCaption('gmailmessage')).toBe('Gmail Message');
+
+    });
+
+    it('should set the values correctly', () => {
+      expect(test.cmp.options.values).toEqual(['foo', 'bar']);
+
+    });
+
+    it('should expand the component correctly', () => {
+      test.cmp.openContainer();
+      expect($$(test.cmp.getCheckboxContainer()).hasClass('coveo-simplefilter-checkbox-container-expanded')).toBe(true);
+
+    });
+
+    it('should collapse the component correctly', () => {
+      test.cmp.closeContainer();
+      expect($$(test.cmp.getCheckboxContainer()).hasClass('coveo-simplefilter-checkbox-container-expanded')).toBe(false);
+    });
+
+    it('should use the correct selected values', () => {
+      test.cmp.selectCheckbox('foo');
+      expect(test.cmp.getSelectedCaptions()).toEqual(['foo']);
+    });
+
+    it('should trigger a query when a checkbox is checked', () => {
+      test.cmp.toggleCheckbox('foo');
+      expect(test.env.queryController.executeQuery).toHaveBeenCalled();
+    });
+
+    it('should set the field in the query', () => {
+      let simulation = Simulate.query(test3.env);
+      expect(simulation.queryBuilder.groupByRequests).toEqual(jasmine.arrayContaining([
+        jasmine.objectContaining({
+          field: '@field'
+        })
+      ]));
+    });
+
+    it('should set the selected values in the query', () => {
+      test.cmp.selectCheckbox('foo');
+      let simulation = Simulate.query(test.env);
+      expect(simulation.queryBuilder.advancedExpression.build()).toEqual('@field==foo');
+    });
+
+    it('should set the right title depending on values selected', () => {
+      test.cmp.selectCheckbox('foo');
+      expect($$($$(test.cmp.root).find('.coveo-simplefilter-selecttext')).text()).toEqual('foo');
+      test.cmp.selectCheckbox('bar');
+      expect($$($$(test.cmp.root).find('.coveo-simplefilter-selecttext')).text()).toEqual(('FooTitleBar'));
+    });
+
+    it('should handle the backdrop correctly', () => {
+      expect($$(test.cmp.root).find('.coveo-dropdown-background')).not.toBe(undefined);
+      test.cmp.openContainer();
+      expect($$($$(test.cmp.root).find('.coveo-dropdown-background')).hasClass('coveo-dropdown-background-active')).toBe(true);
+      test.cmp.closeContainer();
+      expect($$($$(test.cmp.root).find('.coveo-dropdown-background')).hasClass('coveo-dropdown-background-active')).toBe(false);
+    });
+
+    it('should put all simple filters in the same wrapper if there is more than one', () => {
+      expect(test.cmp.element.parentElement).toEqual(test2.cmp.element.parentElement);
```

Comment:
Only place you use `test2`. Could just instance another mock inside this test `anotherSimpleFilter`.